cmd/istioctl/inject.go

```diff
@@ -113,12 +113,24 @@ kubectl get deployment -o yaml | istioctl kube-inject -f - | kubectl apply -f -
 				return err
 			}
 
-			mesh, err := inject.GetMeshConfig(client, namespace, meshConfig)
+			istioMeshConfigMap, mesh, err := inject.GetMeshConfig(client, istioNamespace, meshConfig)
 			if err != nil {
 				return fmt.Errorf("Istio configuration not found. Verify istio configmap is "+
 					"installed in namespace %q with `kubectl get -n %s configmap istio`",
-					namespace, namespace)
+					istioNamespace, istioNamespace)
 			}
+
+			// Temporary hack (few days), until this is properly implemented
+			// https://github.com/istio/pilot/issues/1153
+			_, _, err = inject.GetMeshConfig(client, namespace, meshConfig)
```

Comment:
This is error prone, though perhaps not a huge deal if its very short lived.

Expected usage:
```bash
# Correct
kubectl apply -n NAMESPACE -f <(istioctl kube-inject -n NAMESPACE -f deployment.yaml)
istioctl kube-inject -n NAMESPACE -f deployment.yaml | kubectl appy -n NAMESPACE -f -

# Bug!
kubectl apply -n NAMESPACE -f <(istioctl kube-inject -f deployment.yaml)
istioctl kube-inject -f deployment.yaml | kubectl appy -n NAMESPACE -f -
```

Reply:
Yes, user need to specify the namespace in the istioctl kube-inject command as well, in addition to the kubectl apply (so twice). They will really want to use the new initializer.

Reply:
Not if their cluster admin has disabled initializers.

Reply:
If they disabled initializers they will have to provide the extra params to istioctl kube-inject,